identity-account/src/updates/update.rs

```diff
@@ -0,0 +1,465 @@
+        );
+
+        let public: PublicKey = if let Some(method_private_key) = method_secret {
+          insert_method_secret(storage, did, &location, type_, method_private_key).await
+        } else {
+          storage.key_new(did, &location).await
+        }?;
+
+        let method: IotaVerificationMethod =
+          IotaVerificationMethod::from_did(did.to_owned(), KeyType::Ed25519, &public, location.fragment().name())?;
+
+        state.store_method_generations(location.fragment().clone());
+
+        // We can ignore the result: we just checked that the method does not exist.
+        state.document_mut().insert_method(method, scope);
+      }
+      Self::DeleteMethod { fragment } => {
+        let fragment: Fragment = Fragment::new(fragment);
+
+        // The verification method must exist
+        ensure!(
+          state.document().resolve_method(fragment.identifier()).is_some(),
+          UpdateError::MethodNotFound
+        );
+
+        // TODO: Do we have to ? here?
+        let method_url: IotaDIDUrl = did.to_url().join(fragment.identifier())?;
+        let core_method_url: CoreDIDUrl = CoreDIDUrl::from(method_url.clone());
+
+        // Prevent deleting the last method capable of signing the DID document.
+        let capability_invocation_set = state.document().as_document().capability_invocation();
+        let is_capability_invocation = capability_invocation_set
+          .iter()
+          .any(|method_ref| method_ref.id() == &core_method_url);
+
+        ensure!(
+          !(is_capability_invocation && capability_invocation_set.len() == 1),
+          UpdateError::InvalidMethodFragment("cannot remove last signing method")
+        );
+
+        state.document_mut().remove_method(method_url).unwrap();
+      }
+      Self::AttachMethod {
+        fragment,
+        relationships,
+      } => {
+        let fragment: Fragment = Fragment::new(fragment);
+
+        // TODO: Do we have to ? here?
+        let method_url: IotaDIDUrl = did.to_url().join(fragment.identifier())?;
+
+        // The verification method must exist
+        ensure!(
+          state.document().resolve_method(fragment.identifier()).is_some(),
+          UpdateError::MethodNotFound
+        );
+
+        // The verification method must not be embedded.
+        ensure!(
+          !state
+            .document()
+            .as_document()
+            .verification_relationships()
+            .any(|method_ref| match method_ref {
+              MethodRef::Embed(method) => method.id().fragment() == method_url.fragment(),
+              MethodRef::Refer(_) => false,
+            }),
+          UpdateError::InvalidTargetMethod
+        );
+
+        for relationship in relationships {
+          // We ignore the boolean result: if the relationship already existed, that's fine.
+          state
+            .document_mut()
+            .attach_method_relationship(method_url.clone(), relationship)
+            .map_err(|_| UpdateError::MethodNotFound)?;
```

Comment:
Nit: explicitly ignore the result.

```suggestion
          // Ignore the boolean result: if the relationship already existed, that's fine.
          let _ = state
            .document_mut()
            .attach_method_relationship(method_url.clone(), relationship)
            .map_err(|_| UpdateError::MethodNotFound)?;
```